app/services/foreman/renderer/scope/macros/snippet_rendering.rb

```diff
@@ -10,9 +10,20 @@ module SnippetRendering
             sections only: %w[all provisioning]
           end
 
+          apipie :method, 'Checks if a snippet exitsts in template source, e.g. database' do
+            desc 'Checks if a snippet exists, does not render template.'
+            required :name, String, desc: 'Name of the snippet template to check for'
+            returns [true, false], desc: 'presence of the snippet'
+            example "snippet_exist('motd') # => true"
+            see 'snippet', description: 'Snippet#snippet', scope: Foreman::Renderer::Scope::Macros::SnippetRendering
+          end
+          def snippet_exists(name)
+            source.find_snippet(name).present?
```

Comment:
I don't think this works safely because I think `nil.present?` is not defined.

Reply:
I've done manual testing on all of this and it behaved as I expected/wanted it to for the different scenarios... 🤷